symengine/type_codes.inc

```diff
@@ -1,57 +1,157 @@
-SYMENGINE_ENUM(INTEGER),
-SYMENGINE_ENUM(RATIONAL),
-SYMENGINE_ENUM(COMPLEX),
-SYMENGINE_ENUM(COMPLEX_DOUBLE),
-SYMENGINE_ENUM(REAL_MPFR),
-SYMENGINE_ENUM(COMPLEX_MPC),
-SYMENGINE_ENUM(REAL_DOUBLE),
+#ifndef EXCLUDE_INTEGER
+SYMENGINE_ENUM(INTEGER, Integer)
+#endif
+#ifndef EXCLUDE_RATIONAL
+SYMENGINE_ENUM(RATIONAL, Rational)
+#endif
+#ifndef EXCLUDE_COMPLEX
+SYMENGINE_ENUM(COMPLEX, Complex)
+#endif
```

Comment:
So I would try to replace
```c++
#ifndef EXCLUDE_INTEGER
SYMENGINE_ENUM(INTEGER, Integer)
#endif
#ifndef EXCLUDE_RATIONAL
SYMENGINE_ENUM(RATIONAL, Rational)
#endif
#ifndef EXCLUDE_COMPLEX
SYMENGINE_ENUM(COMPLEX, Complex)
#endif
...
```
with something like
```c++
#define DEFINE_SYMENGINE_ENUM(type, Class) \
#ifndef EXCLUDE_##type \
SYMENGINE_ENUM(type, Class) \
#endif

DEFINE_SYMENGINE_ENUM(INTEGER, Integer)
DEFINE_SYMENGINE_ENUM(RATIONAL, Rational)
DEFINE_SYMENGINE_ENUM(COMPLEX, Complex)
...
```
I don't know if you can nest the `ifndef` inside a `define`, but it would be nice to figure out a solution so that you don't need to repeat all this, and just define the class + type using a simple macro like `DEFINE_SYMENGINE_ENUM(INTEGER, Integer)`.